plugins/httpremote.koplugin/main.lua

```diff
@@ -0,0 +1,226 @@
+local Device =  require("device")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local socket = require("socket")
+local Event = require("ui/event")
+local logger = require("logger")
+local _ = require("gettext")
+
+local HttpRemote = WidgetContainer:extend{
+    name = "httpremote",
+    is_doc_only = true,
+}
+
+function HttpRemote:init()
+    self.port = G_reader_settings:readSetting("httpremote_port", "8080")
+    self.autostart = G_reader_settings:isTrue("httpremote_autostart")
+
+    self.task = function()
+        self:httpListen()
+    end
+
+    if self.autostart then
+        self:start()
+    end
+
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function HttpRemote:start()
+    self.server = socket.bind("*", self.port)
+    self.server:settimeout(0.01) -- set timeout (10ms)
+    logger.dbg("HttpRemote: Server listening on port " .. self.port)
+
+    -- Make a hole in the Kindle's firewall
+    if Device:isKindle() then
+        os.execute(string.format("%s %s %s",
+            "iptables -A INPUT -p tcp --dport", self.port,
+            "-m conntrack --ctstate NEW,ESTABLISHED -j ACCEPT"))
+        os.execute(string.format("%s %s %s",
+            "iptables -A OUTPUT -p tcp --sport", self.port,
+            "-m conntrack --ctstate ESTABLISHED -j ACCEPT"))
+    end
+
+    -- start listening
+    UIManager:scheduleIn(0.01, self.task)
+end
+
+function HttpRemote:stop()
+    if self.server then
+        logger.dbg("HttpRemote: Server stopped.")
+        self.server:close()
+        self.server = nil
+
+        -- stop listening
+        UIManager:unschedule(self.task)
+
+        -- Plug the hole in the Kindle's firewall
+        if Device:isKindle() then
+            os.execute(string.format("%s %s %s",
+                "iptables -D INPUT -p tcp --dport", self.port,
+                "-m conntrack --ctstate NEW,ESTABLISHED -j ACCEPT"))
+            os.execute(string.format("%s %s %s",
+                "iptables -D OUTPUT -p tcp --sport", self.port,
+                "-m conntrack --ctstate ESTABLISHED -j ACCEPT"))
+        end
+    else
+        logger.dbg("HttpRemote: No server running.")
+    end
+end
+
+function HttpRemote:httpListen()
+    if self.server then
+        local client = self.server:accept() -- wait for a client to connect
+        if client then
+            self:onRequest(client) -- handle the request
+            client:close() -- close the connection to the client
+        end
+        UIManager:scheduleIn(0.01, self.task)
+    end
+end
+
+function HttpRemote:onRequest(client)
+    local request = client:receive("*l") -- read the first line of the request
+
+    logger.dbg("HttpRemote: Received request: " .. request)
+    local params_string = request:match("%u+%s+%S+%?([^%s]+)%s+%S+") -- extract params
+    logger.dbg("HttpRemote: Params: " .. params_string)
```

Comment:
It should read like this:
```suggestion
    local params_string = request:match("%u+%s+%S+%?([^%s]+)%s+%S+") -- extract params
    logger.dbg("HttpRemote: Params: ", params_string)
```